pkg/helpers/helpers.go

```diff
@@ -65,10 +65,18 @@ func EncodeSnowflakeID(attributes ...interface{}) string {
 	// is attribute already an object identifier?
 	if len(attributes) == 1 {
 		if id, ok := attributes[0].(sdk.ObjectIdentifier); ok {
-			// remove quotes and replace dots with pipes
-			parts := strings.Split(id.FullyQualifiedName(), ".")
-			for i, part := range parts {
-				parts[i] = strings.Trim(part, `"`)
+			parts := make([]string, 0)
+			switch v := id.(type) {
```

Comment:
Should we also consider cases where a pointer to an identifier is passed? If not then, we'll hit the default branch

Reply:
I will add this.

Reply:
Done.

Reply:
I think the matching cases could be joined but other than that 👍

Reply:
Tried that, and it worked only for AccountObjectIdentifier :(

Reply:
![Uploading Screenshot 2023-12-12 at 16.05.26.png…]()